### ISSUE TITLE
Issue in 2.1.0 Installation

### ISSUE THREAD

**Issue by dsmaldone** [irontec/ivozprovider#467]
Hi guys,
I faced an issue installing artemis 2.1.0
```
[ ok ] Starting mysql (via systemctl): mysql.service.
ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)
ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)
/opt/irontec/ivozprovider/scheme /
Buildfile: /opt/irontec/ivozprovider/scheme/build_prod.xml
 [property] Loading /opt/irontec/ivozprovider/scheme/./build.properties.production

ivozprovider > init:

Execution of target "init" failed for the following reason: /opt/irontec/ivozprovider/scheme/build_prod.xml:15:13: Task exited with code 1

BUILD FAILED
/opt/irontec/ivozprovider/scheme/build_prod.xml:15:13: Task exited with code 1
Total time: 0.2066 seconds

/
Buildfile: /opt/irontec/ivozprovider/scheme/build_prod.xml
 [property] Loading /opt/irontec/ivozprovider/scheme/./build.properties.production

ivozprovider > first-pass-migrate:
```

Seems the issue is in the script generating **build.properties.production** as whatever settings I input in the wizard, I got always the same configuration
```
This dir must contain the local application
build.dir=./

# Credentials for the database migrations
db.host=localhost
db.user=root
db.pass=changeme
db.name=ivozprovider

# paths to programs
progs.mysql=/usr/bin/mysql

application.env=production
```

The _quick workaround_ was to set 'changeme' as password during the installation

I will try to help you fix this in the coming days

Thx for your great job

**Comment by mmadariaga** [comment]
Thanks for reporting

**Comment by dsmaldone** [comment]
Seems it's related and already presented here https://github.com/irontec/ivozprovider/issues/450

**Comment by manfer** [comment]
I don't know if it is the same on the ISO than installing from debian packages, but to be able to test v.2.x I had to use as the mysql root password the default ivozprovider admin password **changeme** or otherwise the installation fails when trying to do the migrations, insert data on the database. So maybe it is hardcoded somewhere.

**Comment by Kaian** [comment]
There must be something quite wrong here.

The provided output is from _phing_, the tool we used in oasis release for managing database migrations. Artemis uses doctrine migrations, so there must be some error in packages or the ISO image.

The artemis branch, doesn't even have those files in [scheme directory](https://github.com/irontec/ivozprovider/tree/artemis/scheme)

Did you install that from the ISO image? Can you provide the output of `dpkg -l | grep ivozprovider`?

Thanks for reporting!

**Comment by dsmaldone** [comment]
So Kaian,
basically, you're right.
I Installed ivoz oasis from ISO and everything worked correctly.
Then I tried to upgrade to artemis (moving the repositories to artemis) but there were a lot of unsatisfied dependencies.

Then I tried to install 2.1.0 from apt (debian packages) without success (missing dependencies) and then from ISO and the installation stucks on "Configuring ivozprovider-scheme"

In summary
- I was not able to install in any way 2.1.0 artemis
- Installing from apt has a bug in mysql root password definition

![screenshot from 2018-05-02 13-09-11](https://user-images.githubusercontent.com/38856598/39520203-0fa78cac-4e0a-11e8-9c3b-d2d02513395c.png)

**Comment by Kaian** [comment]
I will check the ISO for atemis as soon as I can.

The docs for artemis branch aren't still updated, but task is handled in #442 

For the record, artemis repositories are:
```cd /etc/apt/sources.list.d
echo deb http://packages.irontec.com/debian artemis main extra > ivozprovider.list
echo deb http://packages.irontec.com/debian tayler main > klear.list
```

Also, note that 2.x is still on alpha stage and still highly developed, so we encourage the installation for testing and trying what's new more than for production environments.

Thanks!

**Comment by manfer** [comment]
> I don't know if it is the same on the ISO than installing from debian packages, but to be able to test v.2.x I had to use as the mysql root password the default ivozprovider admin password changeme or otherwise the installation fails when trying to insert data on the database. So maybe it is hardcoded somewhere.

Sorry. I have tested just now and it works correctly. So don't pay attention to that comment.

Almost sure it was my fault and forgot to set the password in ivozprovider configuration. The configuration comes just inmediatly after configuring the root mysql password and I probably thought it was not needed to type it again.

**Comment by Kaian** [comment]
I also have successfully tested the installation at least from packages repositories.

As @manfer mentions, I have only configured mysql password twice: first in mysql-server package dialog and then at ivozprovider install menu and the installation ends without problem. Entering the database password at two different screens it's a bit confusing, but that way you can later update the mysql password from mysql cli and run `dpkg-reconfigure ivozprovider` to set the new password to all required files.

Attached is the log, from a fresh debian 9 installation.

[ivozprovider-2.1-install.log](https://github.com/irontec/ivozprovider/files/1967709/ivozprovider-2.1-install.log)

I will try to install 2.1 ISO later. 

Thanks for the feedback!

**Comment by Kaian** [comment]
The above commit should avoid the stuck problem during the ISO installation (it still takes a bit, so be patient).

As we are quite close to the 2.2.0 release, I will fix this directly on the next ISO. If you want to alpha test the changes, they will be in the [isos directory](http://packages.irontec.com/isos/) under the ~alpha tag until release has been closed.

Regards

**Comment by Kaian** [comment]
For the record, it seems I have to update rtpengine dependencies to properly compile its kernel module, so it won't autostart after ISO install. We are aware of the problem and will try to solve it for the release date.

**Comment by Kaian** [comment]
Is this still a problem for 2.2.0 ISOs?

**Comment by dsmaldone** [comment]
Hi Kaian,
just saw the last comment.
Checked and downloaded 

ivozprovider-2.2~2.2.0-artemis-amd64.iso | 2018-05-04 17:51 |
I will test soon
Thx for your work!

Danilo

**Comment by Kaian** [comment]
This should be fixed in 2.3.0 iso
